go/ai/prompt.go

```diff
@@ -23,31 +23,31 @@ import (
 	"github.com/invopop/jsonschema"
 )
 
-// A PromptAction is used to render a prompt template,
-// producing a [GenerateRequest] that may be passed to a [ModelAction].
-type PromptAction = core.Action[any, *GenerateRequest, struct{}]
+// A Prompt is used to render a prompt template,
+// producing a [GenerateRequest] that may be passed to a [Model].
+type Prompt core.Action[any, *GenerateRequest, struct{}]
 
 // DefinePrompt takes a function that renders a prompt template
-// into a [GenerateRequest] that may be passed to a [ModelAction].
+// into a [GenerateRequest] that may be passed to a [Model].
 // The prompt expects some input described by inputSchema.
 // DefinePrompt registers the function as an action,
-// and returns a [PromptAction] that runs it.
-func DefinePrompt(provider, name string, metadata map[string]any, render func(context.Context, any) (*GenerateRequest, error), inputSchema *jsonschema.Schema) *PromptAction {
+// and returns a [Prompt] that runs it.
+func DefinePrompt(provider, name string, metadata map[string]any, render func(context.Context, any) (*GenerateRequest, error), inputSchema *jsonschema.Schema) *Prompt {
```

Comment:
should `inputSchema` be before the `render func`?

Reply:
I guess what I mean is that `render func` should always be last

Reply:
I'll do that in a followup.